packages/dds/tree/src/core/tree/anchorSet.ts

```diff
@@ -268,6 +318,51 @@ export class AnchorSet {
 		return parentPath?.tryGetChild(path.parentField, path.parentIndex);
 	}
 
+	/**
+	 * Returns an equivalent path making as much of it with PathNodes as possible.
+	 * This allows future operations (like fine, track, locate) on this path (and derived ones) to be faster.
+	 *
+	 * @remarks
+	 * Also ensures that any PathNode in the path is from this AnchorSet.
+	 */
+	public internalizePath(path: UpPath): UpPath {
```

Comment:
Added a todo that this class should not be exported.

It needs to be public since it has a test.

Reply:
Can we mark it @internal?

Reply:
(for now)

Reply:
We can't mark AnchorSet `@internal` since (ae-incompatible-release-tags). You can't mark any members of it as `@internal` because of `Warning: (ae-incompatible-release-tags) The symbol "AnchorNode" is marked as @alpha, but its signature references "AnchorEvents" which is marked as @internal`


Also I'm not convinced this API is really something we want to make any more private than the other ones here. Things that want trees and sometimes need anchors may want access to this to help avoid their anchor creation going O(D^2). The current way cursors are implemented (without something like this) for exampela re one of the known perf issues for editable tree.